Yoast/javascript
# Add Arabic external stemmer

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adds and refactors an external stemmer for Arabic
* [Yoast SEO Premium] Adds word forms support for Arabic keyphrase recognition.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Run `yarn test` and make sure that all specs pass
* Add more words to spec file and delete them after testing. For example different verb conjugations that follow different verb models, see https://conjugator.reverso.net/conjugation-rules-model-arabic.html. Make sure that all different forms of the same verb have the same stem.
* In content analysis app:
    * Turn the morphology feature on and add minimum 150 words to the text field
    * Set the locale to "ar"
    * Use `بحث` as the focus keyphrase 
    * Add these other forms of the word in the text field `يبحث, نبحث, بحثت, بحثوا, يبحثن, مبحوث, باحث, البحث, الابحاث, للبحث` and make sure that all these forms are recognized.

## QA test instruction
* In your testing environment, set the language to Arabic
* Make sure to test this functionality in premium
* Add text minimum 150 words
* Use `بحث` as the focus keyphrase 
* Add these other forms of the word in the text field `يبحث, نبحث, بحثت, بحثوا, يبحثن, مبحوث, باحث, البحث, الابحاث, للبحث` and make sure that all these forms are recognized.



## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-550
